prusti-viper/src/encoder/specification_blocks.rs

```diff
@@ -0,0 +1,174 @@
+use std::collections::{BTreeMap, BTreeSet};
+use prusti_rustc_interface::{
+    data_structures::graph::WithSuccessors,
+    middle::{mir, ty::TyCtxt},
+};
+
+use prusti_interface::environment::{
+    is_ghost_begin_marker, is_ghost_end_marker, is_loop_invariant_block,
+    is_marked_specification_block, Procedure,
+};
+
+/// Information about the specification blocks.
+pub(super) struct SpecificationBlocks {
+    /// All blocks that are generated as a result of specification type-checking.
+    specification_blocks: BTreeSet<mir::BasicBlock>,
+    /// Blocks through which specifications are entered.
+    specification_entry_blocks: BTreeSet<mir::BasicBlock>,
+    ghost_blocks: BTreeSet<mir::BasicBlock>,
+    /// A set of blocks containing the loop invariant of a given loop in
+    /// execution order.
+    ///
+    /// FIXME: Add a check that ensure that the blocks are one after another.
```

Comment:
Do you mean that all `body_invariant!(...)` are placed next to each other? If so, the proper fix is rather to encode invariants properly no matter where they are placed.

Reply:
As we now use the `SpecificationBlocks` from the new core proof encoder, this is no longer relevant here.